client/src/containers/Entity/EntityForm.js

```diff
@@ -1,13 +1,18 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { TextField } from '../../components/elements';
+import { useEntityTypes } from './EntityTypesContextProvider';
 
-class EntityForm extends Component {
-  render() {
-    const { withFieldData, entityType } = this.props;
-    const NameField = withFieldData(TextField, 'name');
-    return <NameField label="Name" helperText={`Name of the ${entityType}`} />;
-  }
+function EntityForm({ withFieldData, entityType }) {
+  const { getEntityType } = useEntityTypes();
+  const entity = getEntityType(entityType) || {};
+  const NameField = withFieldData(TextField, 'name');
+  return (
+    <NameField
```

Comment:
Would it be possible to omit this change from this PR, the rest of the changes look great!
I'll create a separate ticket for the name optionality aspect.